tmt/steps/__init__.py

```diff
@@ -863,42 +871,55 @@ def login(context: Any, **kwargs: Any) -> None:
 
     @classmethod
     def plugins(cls, step: Step) -> List['Login']:
+        from tmt.steps.execute import ExecutePlugin
         """ Return list of login instances for given step """
         if not Login._enabled:
             return []
+        if cls._opt('test') and step.name == 'execute':
+            for p in step.phases(classes=(ExecutePlugin,)):
+                p._login_after_test = Login(step, 90)
         return [Login(step, phase) for phase in cls.phases(step)]
 
     def go(self) -> None:
         """ Login to the guest(s) """
-        # Verify possible test result condition
-        count: Dict[str, int] = dict()
-        expected_results: Optional[List[str]] = self.opt('when')
 
-        assert isinstance(self.parent, Step)
-        assert hasattr(self.parent, 'plan') and self.parent.plan is not None
+        if self._enabled_by_results(self.parent.plan.execute.results()):
+            self._login()
 
+    def _enabled_by_results(self, results: List['tmt.base.Result']) -> bool:
+        """ Verify possible test result condition """
+        expected_results: Optional[List[str]] = self.opt('when')
         if expected_results:
-            for expected_result in expected_results:
-                count[expected_result] = len([
-                    result for result in self.parent.plan.execute.results()
-                    if result.result == expected_result])
-            if not any(count.values()):
+            count = Counter([r.result for r in results
+                             if r.result in expected_results])
+            if not len(count):
                 self.info('Skipping interactive shell', color='yellow')
-                return
+                return False
+        return True
 
+    def _login(self, cwd: Optional[str] = None, env: Optional[Dict[str, str]] = None) -> None:
         # Run the interactive command
         commands: List[str] = self.opt('command')
+        env = env or dict()
```

Comment:
Is `env` really needed? IIUIC, it's passed to `guest.execute()` which should be fine with the default value of `env=None` anyway.

Reply:
Actually, this snippet of code causes an error that is attached to the comment. Therefore, I set env to dict in the default case.
```
env=None
guest.execute(command, interactive=True, cwd=cwd, env=env)
```

[error.txt](https://github.com/teemtee/tmt/files/9627204/error.txt)

Reply:
Congratulation, you have discovered a bug in the `local` plugin :) An attempt to fix the issue is [here](https://github.com/teemtee/tmt/pull/1525).

Reply:
Cool :) I'll rebase this branch onto main after the fix is merged. Anyway, I am deleting the line.